### PR TITLE
feat: remove <d2l-navigation-button>

### PR DESCRIPTION
**Pull request by dlockhart**
All usages of this have been replaced with either `<d2l-navigation-button-icon>` or `<d2l-navigation-dropdown-button-*>`! 🎉 

There's [one remaining usage in nova-persona-manager](https://github.com/Brightspace/nova-persona-manager/blob/864bea9145ccb6381994895363dbe2ab906bb882/app/main.js#L230) but that project is still on the Lit 1 version of navigation, which doesn't have any of the new components. They're aware, and I've [created a draft PR](https://github.com/Brightspace/nova-persona-manager/pull/41) that they can just apply when they do the migration.

### CODE REVIEW

File: d2l-navigation-button.js [hunk 1]
```diff
@@ -1,131 +0,0 @@
-import '@brightspace-ui/button/d2l-button-behavior.js';
-import '@brightspace-ui/polymer-behaviors/d2l-focusable-behavior.js';
-import '@brightspace-ui/polymer-behaviors/d2l-id.js';
-import { highlightStyles } from './d2l-navigation-highlight-styles.js';
```
Comment (dlockhart) [inline d2l-navigation-button.js:4]: This and `polymer-behaviors` are still used by `<d2l-navigation-link>` -- so close!

File: package.json [hunk 2]
```diff
@@ -56,7 +55,6 @@
     "stylelint": "^14"
   },
   "dependencies": {
-    "@brightspace-ui/button": "^6",
```
Comment (dlockhart) [inline package.json:59]: 🎉

**Comment by svanherk** [comment]
Looks like Consistent Eval still uses it? https://github.com/Brightspace/consistent-evaluation/blob/master/components/footer/consistent-evaluation-footer.js#L130-L139

**Comment by dlockhart** [comment]
> Looks like Consistent Eval still uses it?

Ugh, my GitHub search for `"<d2l-navigation-button "` has failed me! Without the space you get tons of false positives for the other variants... looks like I'll have to comb through them.

**Comment by svanherk** [comment]
I tried `"d2l-navigation-button>"` instead hoping that the end tag was existent and correct.  But for some reason, the new GitHub search doesn't seem to give me everything.  I found https://search.d2l.dev/xref/Brightspace/status-page/Front-end/src/BrightspaceServices.js?r=2cdc370b#545 with OpenGrok, but that `status-page` repo just won't appear in GitHub search, like I don't have access to it 😕

**Comment by dlockhart** [comment]
Alright, `<d2l-navigation-button>` has been removed [from the status-page](https://github.com/Brightspace/status-page/pull/15), [from consistent-eval](https://github.com/Brightspace/consistent-evaluation/pull/1188) and [from portfolio](https://github.com/Brightspace/folio-app/pull/1561).

I can't find any other references other than the one mentioned in the PR description.

**Review by svanherk — APPROVED** [review]
🎉 🎊 🥳

**Pull request merged**